clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -4020,6 +4020,39 @@ def FMinOp : BinaryFPToFPBuiltinOp<"fmin", "MinNumOp">;
 def FModOp : BinaryFPToFPBuiltinOp<"fmod", "FRemOp">;
 def PowOp : BinaryFPToFPBuiltinOp<"pow", "PowOp">;
 
+def IsFPClassOp : CIR_Op<"is_fp_class"> {
+  let summary = "Corresponding to the `__builtin_fpclassify` builtin function in clang";
+
+  let description = [{
+    The `cir.is_fp_class` operation takes a floating-point value as its first
+    argument and a bitfield of flags as its second argument. The operation
+    returns a boolean value indicating whether the floating-point value
+    satisfies the given flags.
+
+    The flags must be a compile time constant and the values are:
+
+    | Bit # | floating-point class |
+    | -------- | ------- |
+    | 0 | Signaling NaN      |
+    | 1 | Quiet NaN          |
+    | 2 | Negative infinity  |
+    | 3 | Negative normal    |
+    | 4 | Negative subnormal |
+    | 5 | Negative zero      |
+    | 6 | Positive zero      |
+    | 7 | Positive subnormal |
+    | 8 | Positive normal    |
+    | 9 | Positive infinity  |
+  }];
+
+  let arguments = (ins CIR_AnyFloat:$src,
+                       UInt32:$flags);
```

Comment:
Since `$flags` is always a compile-time constant, I recommend invent a new attribute for it. This could also made LLVM lowering easier since `llvm.intr.is.fpclass` accepts an integer attribute as its parameter too.

Reply:
If the value can be a bitwise OR of these values, I recommend the second argument is just an mlir::IntegerAttr. If only one value is possible, an enum kind is the way to go!

Reply:
Done. They can be combined. So I chose I32Attr.